samples/nrf9160/nrf_cloud_mqtt_multi_service/src/connection.c

```diff
@@ -691,6 +702,17 @@ static int setup_cloud(void)
 	struct nrf_cloud_init_param params = {
 		.event_handler = cloud_event_handler
 	};
+
+#if defined(CONFIG_NRF_CLOUD_FOTA_FULL_MODEM_UPDATE)
+	struct dfu_target_fmfu_fdev fmfu_dev_inf = {
+		.size = 0,
+		.offset = 0,
+		.dev = DEVICE_DT_GET_ONE(EXT_FLASH_DEVICE)
+	};
+
+	params.fmfu_dev_inf = &fmfu_dev_inf;
+#endif
+
```

Comment:
If this section stays, then, for readability/stylistic consistency, I think that:

- `if(IS_ENABLED)` would be preferable here instead of straight macros (just let `EXT_FLASH_DEVICE` always be defined, it doesn't need to be conditionally defined)
- `EXT_FLASH_DEVICE` should be named `FULL_MODEM_FOTA_EXT_FLASH_DEVICE` to make its purpose (and where to find it) obvious to beginners
- there should be a comment directing the reader to check `fota_support.h` for its definition

However, it would be nice if this configuration could be controlled entirely in `fota_support.h`, and if this snippet could be avoided altogether.

I see two ways of doing that:

1. Create a static function in `fota_support.h`: `set_full_modem_fota_dev_inf` that either produces the struct or modifies the cloud config struct. But I don't see a way to do that without having to invoke the heap.

2. Make these settings configurable with KConfig and DT overlays somehow, and default to those configured values if they are not passed in via the cloud config struct.

I think 2. is worth pursuing anyways, just for the ease of configuration it would allow

Reply:
An external flash device can be used for more than one purpose, in theory.  For example, MemFault storage, modem trace storage, a file system, and/or P-GPS data (eventually).

Reply:
In this case, it's EXT_FLASH_DEVICE is defined in `fota_support.c` and is exclusively being used for full modem fota

Reply:
moved into fota_support, please review

Reply:
sorry Georges, had to add an `#if defined` back in (fota_support.c)... was getting compile errors due to `DEVICE_DT_GET_ONE(jedec_spi_nor)` when building for a device without external flash.

Reply:
Moving these here, because I somehow managed to write them in the wrong comment thread originally😂


No worries, thanks for trying! Plus, you managed to do it outside of the function body, which is all I really care about anyways, so kudos for that.

I'm pretty sure Zephyr has a ternary operator macro defined somewhere that might let you do what you did as a one-liner, I'll see if I can dig it up...

Such macros do exist, but they are not particularly readable. What you've done maximizes readability, I think

https://docs.zephyrproject.org/latest/kernel/util/index.html in case you decide you want to look at it yourself

IF_ENABLED and COND_CODE_1 would be the macros of interest if you decided to go that route. But I think using obscure macros may actually make things worse for a beginner, whereas standard c macros are pretty readable done the way you've done them